boa_gc/src/internals/gc_box.rs

```diff
@@ -0,0 +1,155 @@
+use crate::Trace;
+use std::cell::Cell;
+use std::ptr::{self, NonNull};
+
+// Age and Weak Flags
+const MARK_MASK: usize = 1 << (usize::BITS - 2);
+const WEAK_MASK: usize = 1 << (usize::BITS - 1);
+const ROOTS_MASK: usize = !(MARK_MASK | WEAK_MASK);
+const ROOTS_MAX: usize = ROOTS_MASK;
+
+pub(crate) struct GcBoxHeader {
```

Comment:
Would it be possible to implement `Debug` here?

Reply:
Done! 😄 Let me know what you think. I was debating putting the isize of `next` into the debug as well but decided against doing it at least for now.